src/Errors.sol

```diff
@@ -0,0 +1,295 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+/// @title Errors
+/// @notice Centralized library for custom error definitions across the protocol
+/// @dev Convention: For any error comparing two values, always pass the expected value first, followed by the actual value
+library Errors {
+    /// @notice Rail does not exist or is beyond its last settlement after termination
+    /// @param railId The ID of the rail
+    error RailInactiveOrSettled(uint256 railId);
+
+    /// @notice Only the rail client can perform this action
+    /// @param expected The expected client address
+    /// @param caller The actual caller address
+    error OnlyRailClientAllowed(address expected, address caller);
+
+    /// @notice Only the rail operator can perform this action
+    /// @param expected The expected operator address
+    /// @param caller The actual caller address
+    error OnlyRailOperatorAllowed(address expected, address caller);
+
+    /// @notice Only the rail participant (client, operator, or recipient) can perform this action
+    /// @param expectedFrom The expected client address
+    /// @param expectedOperator The expected operator address
+    /// @param expectedTo The expected recipient address
+    /// @param caller The actual caller address
+    error OnlyRailParticipantAllowed(address expectedFrom, address expectedOperator, address expectedTo, address caller);
+
+    /// @notice Rail is already terminated
+    /// @param railId The ID of the rail
+    error RailAlreadyTerminated(uint256 railId);
+
+    /// @notice Rail is not terminated, but the action requires a terminated rail
+    /// We can pass railId here but most functions pass rail but we cannot get railId from it
+    error RailNotTerminated();
+
+    /// @notice The rail does not exist
+    error RailDoesNotExist();
+
+    /// @notice The provided address is zero, which is not allowed
+    /// @param varName The name of the variable that was expected to be non-zero
+    error ZeroAddressNotAllowed(string varName);
+
+    /// @notice The rail is inactive
+    /// @param railId The ID of the rail that is inactive
+    error RailInactive(uint256 railId);
```

Comment:
This appears to be only used in exactly the same scenario as `RailInactiveOrSettled`, i.e. the `from` field is address(0).  You should consolidate the two.